React/Views/RCTView.m

```diff
@@ -1380,6 +1380,70 @@ - (void)drawFocusRingMask
 #pragma mark - macOS Event Handler
 
 #if TARGET_OS_OSX
+- (void)resetCursorRects
+{
```

Comment:
I think you might need to call `[self discardCursorRects]` here before adding new ones below.

Reply:
Also, may I recommend having a `NSCursor *cursor` variable that can get set in the switch below, then afterwards add it to reduce some of the repetition here?
```
if (cursor) {
  [self addCursorRect:self.bounds cursor:cursor];
}
```

Reply:
I add one line of code: `[super resetCursorRects];`
It will stable display for cursor.